okhttp/src/main/java/okhttp3/internal/cache/DiskLruCache.kt

```diff
@@ -13,371 +13,350 @@
+    initialized = true
   }
 
-  private void readJournal() throws IOException {
-    try (BufferedSource source = Okio.buffer(fileSystem.source(journalFile))) {
-      String magic = source.readUtf8LineStrict();
-      String version = source.readUtf8LineStrict();
-      String appVersionString = source.readUtf8LineStrict();
-      String valueCountString = source.readUtf8LineStrict();
-      String blank = source.readUtf8LineStrict();
-      if (!MAGIC.equals(magic)
-          || !VERSION_1.equals(version)
-          || !Integer.toString(appVersion).equals(appVersionString)
-          || !Integer.toString(valueCount).equals(valueCountString)
-          || !"".equals(blank)) {
-        throw new IOException("unexpected journal header: [" + magic + ", " + version + ", "
-            + valueCountString + ", " + blank + "]");
+  @Throws(IOException::class)
+  private fun readJournal() {
+    fileSystem.source(journalFile).buffer().use { source ->
+      val magic = source.readUtf8LineStrict()
+      val version = source.readUtf8LineStrict()
+      val appVersionString = source.readUtf8LineStrict()
+      val valueCountString = source.readUtf8LineStrict()
+      val blank = source.readUtf8LineStrict()
+
+      if (MAGIC != magic ||
+          VERSION_1 != version ||
+          appVersion.toString() != appVersionString ||
+          valueCount.toString() != valueCountString ||
+          blank.isNotEmpty()) {
+        throw IOException(
+            "unexpected journal header: [$magic, $version, $valueCountString, $blank]")
       }
 
-      int lineCount = 0;
+      var lineCount = 0
       while (true) {
         try {
-          readJournalLine(source.readUtf8LineStrict());
-          lineCount++;
-        } catch (EOFException endOfJournal) {
-          break;
+          readJournalLine(source.readUtf8LineStrict())
+          lineCount++
+        } catch (_: EOFException) {
+          break // End of journal.
         }
       }
-      redundantOpCount = lineCount - lruEntries.size();
+
+      redundantOpCount = lineCount - lruEntries.size
 
       // If we ended on a truncated line, rebuild the journal before appending to it.
       if (!source.exhausted()) {
-        rebuildJournal();
+        rebuildJournal()
       } else {
-        journalWriter = newJournalWriter();
+        journalWriter = newJournalWriter()
       }
     }
   }
 
-  private BufferedSink newJournalWriter() throws FileNotFoundException {
-    Sink fileSink = fileSystem.appendingSink(journalFile);
-    Sink faultHidingSink = new FaultHidingSink(fileSink) {
-      @Override protected void onException(IOException e) {
-        assert (Thread.holdsLock(DiskLruCache.this));
-        hasJournalErrors = true;
+  @Throws(FileNotFoundException::class)
+  private fun newJournalWriter(): BufferedSink {
+    val fileSink = fileSystem.appendingSink(journalFile)
+    val faultHidingSink = object : FaultHidingSink(fileSink) {
```

Comment:
TIOLI: Might be a good time to make FaultHidingSink a final class and take an exception handler :)

```
val faultHidingSink = FaultHidingSink(fileSink) { ex ->
   assert(Thread.holdsLock(this@DiskLruCache))
   hasJournalErrors = true
}
```

Reply:
Yeah, I like that. Done.